presto-hive/src/main/java/com/facebook/presto/hive/metastore/glue/GlueHiveMetastore.java

```diff
@@ -601,6 +611,25 @@ public void dropColumn(String databaseName, String tableName, String columnName)
     }
 
     private List<Partition> getPartitions(String databaseName, String tableName, String expression)
+    {
+        List<Partition> partitions;
+        if (numSegments <= 1) {
```

Comment:
`numSegments < 1` should not be allowed (and `getMaxGlueConnections` is `@Min(1)` so you can safely reject anything lower than than in the ctor)